Articles/Blog/wikize_refs.py

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/env python
+
+# run ./wikize-refs.py --help for documentation
+
+def usage():
+    return \
+"""
+%prog <infile.md>
+
+Renumbers 1...N and re-formats, slightly, footnotes and a series of
+reference style links in a GitHub Markdown file so that the article's
+footnote links behave more Wikipedia-like. In the new file, the original
+reference style link definitions are copied over but commented out by
+bracketing in <!--- and --->. The renumbered and re-formatted links are
+bi-level. Footnotes in the content link to entries in a table of 
+references at the bottom of the document. Items in the table of references
+link off-page to their intended destinations. The resulting file is still
+GitHub flavored Markdown with a minimal amount of embedded HTML.
+For more information, see https://github.com/betterscientificsoftware/betterscientificsoftware.github.io/blob/master/Articles/Blog/ReferencesInMarkdownHybridApproach.md
+
+By default, infile.md is moved to infile.src.md and the new file is
+given the name infile.md and is made read-only. However, you can set
+the name of the output file instead and disable the read-only setting.
+By default, failed sanity checks cause an abort. However, this can
+be disabled by setting them to warn.
+
+To process a file...
+
+    ./wikize_refs.py foo.md
+
+...moves foo.md to foo.src.md and the new file is foo.md
```

Comment:
I think this is confusing to be moving file names around like this.  I would not rename any files.  I would just make the convention to be:

```
wikize_refs.py <basename>
```

and by default it would be assumed the input file `<basename>.src.md` exists and will be read and the default output file would be named `<basename>.md`.

But if the user gives the `--outfile` (`-o`) option, then the user can call with:

```
wikize_refs.py <basename> -o <basename>-wikize.md
```

where it is assumed that the input file is `<basename>.src.md` or the user can be explicit with the input and output with:

```
wikize_refs.py <basename>.src.md -o <basename>.md
```

or with:

```
wikize_refs.py <basename>.md -o <basename>-wikize.md
```

or whatever.

The rule would be if the input name `<inputfile>` from `wikizize_refs.py <inputfile> ...` does not have the `.md` extension, the input file is assumed to be `<inputfile>.src.md`.  But if it does have the `.md` extension then `<inputfile>` is used as the input file name with no name changes.

This would be simpler in the implementation and less surprising the users I think.  And this is kind of how latex works, right?  If you call:

```
$ pdflatex <some_name>
```

it is assumed the input file is called `<some_name>.tex` and the output file that is generated will be called `<some_name>.pdf`.  Nice a simple.  But in our case, the input and output files both are just Markdown and therefore should have the `*.md` extension since that will allow GitHub to format them properly when views on GitHub.

What do you think about that?

Reply:
Ok, I guess I misunderstood your previous comments on this. I have adjusted the logic now.